### PR TITLE
improve README

### PR DESCRIPTION
**Pull request by lrytz**

### CODE REVIEW

**Comment by lrytz** [comment]
i will create a backport PR for 2.11.x once this is merged, that's easier given that there would be merge conflicts when merging forward from 2.11.x to 2.12.x.

**Comment by lrytz** [comment]
Review by anyone interested :) @SethTisue @adriaanm @retronym @som-snytt @milessabin @janekdb

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -117,14 +117,30 @@ See the [scala-jenkins-infra repo](https://github.com/scala/scala-jenkins-infra)
...
-Once all these conditions are met, and we agree with the change (we are available on scala-internals to discuss this beforehand, before you put in the coding work!), we will merge your changes.
+Once all these conditions are met, and we agree with the change (we are available on
+gitter or scala-internals to discuss this beforehand, before you put in the coding work!),
+we will merge your changes.
+
+We use the following labels:
+
+Label           | Description
+--------------- | -----------
+`reviewed`      | automatically added by scabot when a comment prefixed with LGTM is posted
```
Comment (janekdb) [inline CONTRIBUTING.md]: There are some other labels which could be included at the cost of diluting the focus on these four.
- assistance-appreciated
- WIP

https://github.com/scala/scala/labels

**Comment by milessabin** [comment]
This is really helpful, thank you! :+1: 

FWIW, at the Typelevel hackday on Saturday we found just two JIRA issues with the low hanging fruit label. One was the one we tackled, which really was low hanging. The other was one of @fommil's relating to classloaders/file locks on Windows which I think he was forced to fix by monkey patching the JDK, so not low hanging fruit by any stretch of the imagination.

**Comment by fommil** [comment]
@milessabin the one I am thinking of (SI-9632) probably is low hanging fruit if fixed in the compiler. Working around it required visiting a few levels of Dante's Inferno though :stuck_out_tongue_closed_eyes: 
- https://github.com/fommil/class-monkey
- https://issues.scala-lang.org/browse/SI-9682
- https://issues.scala-lang.org/browse/SI-9632

`class-monkey` is a workaround for SI-9682, and ensime has `ensime-monkeys` for SI-9632 but I still need a way to monkey patch the compiler via sbt to get both fixes in `sbt`. Also, I think if I can make SI-9632 use `URLClassLoader` (plus directory listing) then it will be cached by `class-monkey` and give a performance boost (on Windows) by hitting heap caches instead of hitting disk.

File: README.md [hunk 2]
```diff
@@ -5,208 +5,262 @@ This is the official repository for the [Scala Programming Language](http://www.
...
-      - [2.12.x](http://www.scala-lang.org/files/archive/nightly/2.12.x/)
-
 # Repository structure
 
 ```
 scala/
-+--build.xml                 The main Ant build script, see also under src/build.
-+--pull-binary-libs.sh       Pulls binary artifacts from remote repository.
++--build.sbt                 The main sbt build script
++--build.xml                 The deprecated Ant build script, see also under src/build.
```
Comment (janekdb) [inline README.md]: If `src/build` is mentioned here then it should also be explained what the contents are and why a contributor might be interested.

File: src/intellij/README.md [hunk 3]
```diff
@@ -33,17 +33,51 @@ When switching between 2.11.x and 2.12.x, make sure to run `sbt intellij`.
...
 
-Compiling, running, JUnit tests and debugging should all work.
-You can work on the compiler, the standard library, and other components as well.
+To debug the Scala codebase you can also use "Remote" debug configuration and pass
+the corresponding arguments to the jvm running the compiler / program.
 
-Note that compilation within IntelliJ is performed in a single pass.
-The code is compiled using the "STARR" (stable reference) compiler, as specified by `starr.version` in `versions.properties`.
-This is consistent with the sbt build.
+To run the compiler create an "Application" configuration with
```
Comment (janekdb) [inline src/intellij/README.md:68]: Possible future improvement would be to share these run configurations so they are written out to `.idea/runConfigurations`, add them to the repo and copy them over during the IDEA setup.

File: CONTRIBUTING.md (hunk 1 above)
Comment (lrytz) [inline CONTRIBUTING.md]: hah, i did add wip but forgot to merge / push that change :) will also add the other one.

**Comment by janekdb** [comment]
Not for this PR but it would be good to adopt the branching naming conventions recommended on http://www.scala-lang.org/contribute/hacker-guide.html#branch as a good practice to inspire other contributors :)

**Comment by lrytz** [comment]
i personally didn't follow the branch naming conventions so far :) i don't actually understand why it matters / helps

**Comment by lrytz** [comment]
Pushed a new version that addresses the feedback.

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -117,14 +117,32 @@ See the [scala-jenkins-infra repo](https://github.com/scala/scala-jenkins-infra)
 
 ### Pass code review
 
-Your PR will need to be assigned to one or more reviewers.  You can suggest reviewers yourself; if you're not sure, see the list in [README.md](README.md) or ask on scala-internals.
+Your PR will need to be assigned to one or more reviewers. You can suggest reviewers
+yourself; if you're not sure, see the list in [README.md](README.md) or ask on gitter
```
Comment (fommil) [inline CONTRIBUTING.md:121]: github flavoured markdown will respect these newlines, so this might render weirdly.

Comment (lrytz) [inline CONTRIBUTING.md:121]: i don't think so, check the preview https://github.com/lrytz/scala/blob/c8b618d616349d3515888d8d85e85689bfae7285/CONTRIBUTING.md

Comment (fommil) [inline CONTRIBUTING.md:121]: oh awesome, they've changed that. Used to always catch me out.

**Comment by lrytz** [comment]
@janekdb LGTY? I think it would be nice to merge this soon, as @milessabin would like to link to it from a blog post that is supposed to appear soon. These READMEs are living documents anyway, so further improvements can (and will) always come in.

**Comment by milessabin** [comment]
Agreed ... this is documentation, so as long as there's nothing dangerously misleading in here, then I think it would make sense to get it out as soon as possible and iterate on it later.

**Comment by janekdb** [comment]
LGTM

**Comment by lrytz** [comment]
fixed another typo

**Pull request merged**